server/services/jobSchedulerService.js

```diff
@@ -0,0 +1,87 @@
+const schedule = require('node-schedule')
+const advisoryLock = require('advisory-lock').default
+const moment = require('moment')
+const config = require('../config')
+const logger = require('../../log.js')
+const { onceOnly } = require('./jobs/jobUtils')
+
+module.exports = function createJobSchedulerService(notificationJobs) {
+  const connectionString = `postgres://${config.db.username}:${config.db.password}@${config.db.server}:5432/${
```

Comment:
Silly question but is the port guaranteed?

Reply:
It is (via docker locally, or terraform in AWS).

Still, pulled it out to common config to remove duplication

Reply:
Also cleaned up env var names and made it possible to start the app with all scheduled jobs off, then start them individually when needed

https://github.com/ministryofjustice/licences/pull/885/commits/62d22962eeaca9ac87816c52e31f1d112bad70c3

Reply:
What happens if the env vars are incorrect or not supplied and the connection to Postgresql is not able to be made? Is there an alert / log / fatal error caught?

Reply:
It's the same as the app itself, so the healthchecks would have already failed and application startup wouldn't complete.

Having said that, the databse lock code should be separate from the scheduling code, and the error logging should be better so I'm making some changes.